dashboard/app/controllers/code_review_comments_controller.rb

```diff
@@ -58,6 +58,11 @@ def project_comments
       storage_app_id: @storage_app_id
     ).order(:created_at)
 
+    # Keep teacher comments private between project owner and teacher.
+    if CodeReviewComment.user_is_peer_of_project_owner?(@project_owner, current_user)
+      @project_comments = @project_comments.reject {|comment| comment.commenter.teacher?}
+    end
```

Comment:
Does this only happen if the user is a peer of the project owner? Do we want to keep teacher comments hidden otherwise too, if a student is a not a peer but still viewing the project? Or is that not a valid scenario?

Reply:
I think so? My thought was that the broader access case is handled by our controller action permissions:

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/ability.rb#L94
https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/code_review_comment.rb

And that we'd just filter out the teacher comments if you are a peer of the project owner?

Reply:
I did iterate over permissioning a few times though, so plz let me know if there's a case I'm not seeing.

Reply:
Ahh okay gotcha - yeah my only suggestion may be that just to be safe we can invert this logic to always strip teacher comments unless project_owner == current_user (right?)

Reply:
Would it be if project_owner == current_user or current_user is the teacher?

Reply:
Yup I think I missed the case Molly is describing, and I think your pattern is clearer Sanchit as well. Will update!